app/src/main/java/com/example/gabrielsaruhashi/ramp/activities/TGuideListMapActivity.java

```diff
@@ -135,4 +99,59 @@ protected void loadMap(GoogleMap googleMap) {
         }
     }
 
+    public void onSwitchView (View view) {
+        if (isMapView) {
+            // trigger list view
+            FragmentTransaction ft = getFragmentManager().beginTransaction();
+            mapFragment.getView().setVisibility(View.INVISIBLE);
+            ft.replace(R.id.fragment_placeholder, PlacesFragment.newInstance(places));
+            ft.commit();
+            isMapView = false;
+        } else {
+            // trigger map view
+            FragmentTransaction ft = getFragmentManager().beginTransaction();
+            ft.hide(placesFragment);
+            ft.commit();
+            // load map fragment
+            FragmentManager fm = getSupportFragmentManager();
+            mapFragment = SupportMapFragment.newInstance();
+            fm.beginTransaction().replace(R.id.fragment_placeholder, mapFragment).commit();
+            Two11Client newClient = new Two11Client();
+            newClient.search("enfield", new JsonHttpResponseHandler() {
+                public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
+                    try {
+                        JSONArray placesJson = response.getJSONArray("results");
+                        places.clear();
+                        places.addAll(Places.fromJson(placesJson));
+                        //TODO: YOYO MOVE THIS TO ASYNC
```

Comment:
can you change the comment to ( // TODO (masayukinagase) instead of doing this try / catch, move it to an AsyncTask